src/pages/admin/materiaalaanvragen/index.tsx

```diff
@@ -1,25 +1,275 @@
+	});
+
+	const { data: maintainers } = useGetContentPartners();
+
+	const getMaintainers = () => {
+		const arr = [] as MenuItemInfo[];
+		arr.push({
+			id: 'ALL',
+			label: CP_MATERIAL_REQUEST_TYPE_FITLER_RECORD[CP_MATERIAL_REQUESTS_FILTER_ALL_ID],
+		});
+		maintainers?.items.map((maintainer) => {
+			arr.push({ id: maintainer.id, label: maintainer.name });
+		});
+		return arr;
+	};
+
+	const { data: currentMaterialRequestDetail, isFetching: isLoading } = useGetMaterialRequestById(
+		currentMaterialRequest?.id || null
+	);
+
+	const noData = useMemo((): boolean => isEmpty(materialRequests?.items), [materialRequests]);
+
+	const sortFilters = useMemo(
+		(): SortingRule<{ id: MaterialRequestKeys; desc: boolean }>[] => [
+			{
+				id: filters.orderProp,
+				desc: filters.orderDirection !== OrderDirection.asc,
+			},
+		],
+		[filters]
+	);
+
+	const onSortChange = (
+		orderProp: string | undefined,
+		orderDirection: OrderDirection | undefined
+	): void => {
+		if (filters.orderProp !== orderProp || filters.orderDirection !== orderDirection) {
+			setFilters({
+				...filters,
+				orderProp,
+				orderDirection,
+				page: 1,
+			});
+		}
+	};
+
+	const onPageChange = (pageZeroBased: number, gotoPage: (i: number) => void): void => {
+		gotoPage(pageZeroBased);
+		setFilters({
+			...filters,
+			page: pageZeroBased + 1,
+		});
+	};
+
+	const renderPagination = ({ gotoPage }: { gotoPage: (i: number) => void }): ReactNode => (
+		<PaginationBar
+			className="u-mt-16 u-mb-16"
+			start={Math.max(0, filters.page - 1) * ADMIN_MATERIAL_REQUESTS_TABLE_PAGE_SIZE}
+			total={materialRequests?.total || 0}
+			count={ADMIN_MATERIAL_REQUESTS_TABLE_PAGE_SIZE}
+			onPageChange={(pageZeroBased: number) => onPageChange(pageZeroBased, gotoPage)}
+		/>
+	);
+
+	const renderEmptyMessage = (): ReactNode =>
+		tHtml('pages/admin/materiaalaanvragen/index___geen-materiaalaanvragen');
+
+	const onRowClick = (evt: MouseEvent<HTMLTableRowElement>, row: Row<MaterialRequest>) => {
+		setCurrentMaterialRequest(row.original);
+		setIsDetailBladeOpen(true);
+	};
+
+	const renderDetailBlade = () => {
+		return (
+			<MaterialRequestDetailBlade
+				isOpen={!isLoading && isDetailBladeOpen}
+				onClose={() => setIsDetailBladeOpen(false)}
+				currentMaterialRequestDetail={currentMaterialRequestDetail}
+			/>
+		);
+	};
+
+	const renderContent = (): ReactNode => {
+		return (
+			<Table<MaterialRequest>
+				className="u-mt-24 p-material-requests__table"
+				options={{
+					columns: getAdminMaterialRequestTableColumns(),
+					data: materialRequests?.items || [],
+					initialState: {
+						pageSize: ADMIN_MATERIAL_REQUESTS_TABLE_PAGE_SIZE,
+						sortBy: sortFilters,
+					} as TableState<MaterialRequest>,
+				}}
+				sortingIcons={sortingIcons}
+				pagination={renderPagination}
+				onSortChange={onSortChange}
+				onRowClick={onRowClick}
+			/>
+		);
+	};
+
+	const onTypeClick = (id: string | number): void => {
```

Comment:
Same here, check out `onTypeClick` in `/src/pages/beheer/materiaalaanvragen/index.tsx` and make sure to use the same implementation. If something's not clear, ping me.

Reply:
In the end you will have somehting like this:
```
const showAll = id === ADMIN_MATERIAL_REQUESTS_FILTER_ALL_ID;
		setTypeDropdownLabel(ADMIN_MATERIAL_REQUEST_TYPE_FITLER_RECORD[id]);
		setIsTypeDropdownOpen(false);

		setFilters({
			type: showAll ? undefined : `${id}`,
			page: 1,
		});
```